### DISCUSSION TITLE
Tutorial error: tox publish invalid value for Version

### DISCUSSION THREAD

**Discussion by austinmw** [pyscaffold/pyscaffold#672] [Q&A] [answered]
Hi, I'm completely new to PyScaffold and was following the tutorial at: https://pyscaffold.org/en/stable/usage.html

I ran the steps in the video, and then attempted to additionally run `tox -e publish`

This command errored with:

> HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/
>          '0.1.0.post1.dev0+gd01fff2.d20221018' is an invalid value for Version. Error: Can't use PEP 440 local versions.

I ran the video's command of `git tag v0.1.0`, so I'm unsure why additional version information is being added. Is there a step I can change in interactive mode or before attempting to publish so that this will work?

**Comment by abravalheri** [comment, marked as answer]
Hi @austinmw this usually happens when you have uncommited changes in the working repository.

If you commit everything before running `git tag`, you should get back a clean version: https://pyscaffold.org/en/stable/faq.html#best-practices-and-common-errors-with-version-numbers.

**Comment by austinmw** [comment]
Thanks I will try that! One more related newbie question if you don't mind, what is the best point in the process to actually create the new repository in GitHub (if using this)? And is there any functionality built in to do this?

**Comment by abravalheri** [comment]
> One more related newbie question if you don't mind, what is the best point in the process to actually create the new repository in GitHub (if using this)?

That will depend on when you want to start pushing your code (or having an external backup), there is no right answer. But I usually would say that is good to get into the habit of using a remote repository as soon as possible.

You can create an empty repository even before running `putup`, then you just have to remember to periodically push your changes.

> And is there any functionality built in to do this?

I am afraid not. PyScaffold will not create a repository for you.
PyScaffold can create Continuous Integration files automatically that integrate with Github (e.g. via the `--github-actions` flag), but not the repository itself
